src/actions/WindowActions.js

```diff
@@ -761,6 +766,113 @@ export function fetchTopActions(windowType, docId, tabId) {
   };
 }
 
+const getAPIUrl = function({ windowId, docId, tabId, rowId, path }) {
+  let documentId = docId;
+
+  if (!docId && rowId) {
+    documentId = rowId[0];
+  }
+
+  return `${config.API_URL}/window/${windowId}${
+    documentId ? `/${documentId}` : ''
+  }${rowId && tabId ? `/${tabId}/${rowId}` : ''}/${path}`;
+};
+
+export function callAPI({ windowId, docId, tabId, rowId, target, verb, data }) {
+  return (dispatch) => {
+    const parentUrl = formatParentUrl({ windowId, docId, rowId, target });
+    if (!parentUrl) return;
+    dispatch(updateCommentsPanelOpenFlag(true));
+    // -- GET call - adapt shape as needed
+    if (verb === 'GET') {
+      return axios.get(parentUrl).then(async (response) => {
+        const data = response.data;
+        let rowData = null;
+
+        if (docId && rowId) {
+          if (rowId.length === 1) {
+            const childUrl = getChangelogUrl(windowId, docId, tabId, rowId);
+            rowData = await axios.get(childUrl).then((resp) => resp.data);
+          }
+        }
+
+        if (rowData) {
+          data.rowsData = rowData;
+        }
+        // update corresponding target in the store - might be adapted for more separated entities
+        if (target === 'comments') {
+          dispatch(updateCommentsPanel(data));
+        }
+        // -- end updating corresponding target
+        dispatch(
+          initDataSuccess({
+            data,
+            docId,
+            scope: 'modal',
+          })
+        );
+      });
+    }
+
+    // -- actions dispatched on POST below
+    if (verb === 'POST') {
+      const dataToSend = preFormatPostDATA({ target, postData: { txt: data } });
+      return axios.post(parentUrl, dataToSend).then(async (response) => {
+        const data = response.data;
+        if (target === 'comments') {
+          dispatch(
+            callAPI({
+              windowId,
+              docId,
+              tabId,
+              rowId,
+              target,
+              verb: 'GET',
+            })
+          );
+          dispatch(updateCommentsPanelTextInput('')); // clear the input in the form
+        }
+        return data;
+      });
+    }
+  };
+}
+
+/**
+ * Formats the url for the api call
+ */
+export function formatParentUrl({ windowId, docId, rowId, target }) {
```

Comment:
Will we use this in cases other than comments ?

Reply:
yes it can be reused. Will put it somewhere under the api/